src/main/java/com/gradle/GitMetadata.java

```diff
@@ -0,0 +1,42 @@
+package com.gradle;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static com.gradle.Utils.execAndGetStdOut;
+
+class GitMetadata {
```

Comment:
I like having a class here encapsulate the logic for fetching the gitmetadata but it seems to me that the encapsulation is not fully complete and requires the caller to know the implementation details. for example you must know you need to pass either the `cmd` args or the value supplier depending on whether git is installed or not.

Reply:
That's a good point, actually I made a change to be more flexible and silently accept the case: git is installed but no cmd is provided.